src/formatDate.js

```diff
@@ -9,6 +9,67 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const partsOfOldDate = date.split(fromFormat[fromFormat.length - 1]);
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const currentDate = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        currentDate.yearYYYY = partsOfOldDate[i];
+        currentDate.yearYY = partsOfOldDate[i][2] + partsOfOldDate[i][3];
+        break;
+
+      case 'YY':
+        currentDate.yearYY = partsOfOldDate[i];
+
+        if (+partsOfOldDate[i] < 30) {
+          currentDate.yearYYYY = `20${partsOfOldDate[i]}`;
+        } else {
+          currentDate.yearYYYY = `19${partsOfOldDate[i]}`;
+        }
+        break;
+
+      case 'MM':
+        currentDate.month = partsOfOldDate[i];
+        break;
+
+      case 'DD':
+        currentDate.day = partsOfOldDate[i];
+        break;
+
+      default:
+        break;
+    }
```

Comment:
It's a good practice to avoid magic numbers. The `- 1` in `fromFormat[fromFormat.length - 1]` is not immediately clear to readers. You could define a variable for the separator to improve readability.